components/certificate-mgt/org.wso2.carbon.identity.certificate.management/src/main/java/org/wso2/carbon/identity/certificate/management/dao/ApplicationCertificateManagementDAO.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.certificate.management.dao;
+
+import org.wso2.carbon.identity.certificate.management.exception.CertificateMgtException;
+import org.wso2.carbon.identity.certificate.management.model.Certificate;
+
+/**
+ * This interface defines the Application Certificate Management DAO.
+ * This supports using auto-incremented IDs for certificate management operations in application-mgt.
+ *
+ * @deprecated It is recommended to use {@link CertificateManagementDAO}, which supports operations with UUID.
+ */
+@Deprecated
+public interface ApplicationCertificateManagementDAO {
```

Comment:
Why do we add a new class as deprecated?

Reply:
To facilitate certificate management within the application-mgt component, a new certificate-mgt central component has been introduced. This design ensures that only the certificate-mgt component accesses the IDN_CERTIFICATE table, as discussed with @malithie , @darshanasbg , and @omindu. Separate classes were added for operations using integer IDs to avoid immediate data migration for existing entries in both the IDN_CERTIFICATE and SP_META_DATA tables (where certificate IDs are referenced).

These new classes have also been marked as deprecated to streamline their removal in the future. Additionally, there is a requirement to provide a REST API for certificate management where it gives a broader context to decide on the migration approach, allowing the deprecated classes to be phased out without doing data migration at this stage.